wagtail/migrations/0079_fix_log_entry_json_timestamps.py

```diff
@@ -0,0 +1,129 @@
+# Generated by Django 3.2.6 on 2022-11-09 07:50
+
+from datetime import datetime
+
+import django.core.serializers.json
+from django.conf import settings
+from django.db import migrations, models
+from django.utils import timezone, dateparse
+
+
+def legacy_to_iso_format(date_string, tz=None):
+    dt = datetime.strptime(date_string, "%d %b %Y %H:%M")
+    if settings.USE_TZ:
+        dt = timezone.make_aware(dt, timezone.utc if tz is None else tz)
+        dt = timezone.localtime(dt, timezone.utc)
+    # We return the datetime object, so DjangoJSONEncoder will serialize it accordingly.
+    return dt
+
+
+def iso_to_legacy_format(date_string, tz=None):
+    dt = dateparse.parse_datetime(date_string)
+    if dt is None:
+        raise ValueError("date isn't well formatted")
+    if settings.USE_TZ:
+        dt = timezone.localtime(dt, timezone.utc if tz is None else tz)
+    return dt.strftime("%d %b %Y %H:%M")
+
+
+def migrate_logs_with_created_only(model, converter):
+    for item in model.objects.filter(
+        action__in=["wagtail.revert", "wagtail.rename", "wagtail.publish"]
+    ).only("data"):
+        try:
+            # If a previous_revision was available, the data contains "revision" with
+            # its created date.
+            # Also, there are "wagtail.publish" logs, which don't set data at all.
+            created = item.data["revision"]["created"]
+            # "created" is set to the previous revision's created_at, which is set
+            # to UTC by django.
+            item.data["revision"]["created"] = converter(created)
+        except ValueError:
+            # TODO TBD: log error?
```

Comment:
I was not completely sure, if we should log/fail on ValueErrors (e.g. unexpected date format), or just ignore them. Looking at it again, I now think we should fail on ValueErrors.

As far as I can see, eg. with the introduction of audit logging in https://github.com/wagtail/wagtail/commit/d7d63c3a, the log timestamps have always used the format `%d %b %Y %H:%M`, I haven't found any other format yet. So it should not fail, I think.